packages/astro/src/core/config/schema.ts

```diff
@@ -271,6 +271,43 @@ export const AstroConfigSchema = z.object({
 				.boolean()
 				.optional()
 				.default(ASTRO_CONFIG_DEFAULTS.experimental.optimizeHoistedScript),
+			i18n: z.optional(
+				z
+					.object({
+						defaultLocale: z.string(),
+						locales: z.string().array(),
+						fallback: z.record(z.string(), z.string().array()).optional(),
+						detectBrowserLanguage: z.boolean().optional().default(false),
+					})
+					.optional()
+					.refine((i18n) => {
+						if (i18n) {
+							const { defaultLocale, locales, fallback } = i18n;
+							if (locales.includes(defaultLocale)) {
```

Comment:
```suggestion
							if (!locales.includes(defaultLocale)) {
```

Looks like an incorrect check